SmartTutor/Recommenders/KnowledgeBasedRecommender.cs

```diff
@@ -20,5 +27,39 @@ public List<LearningObject> FindEducationalContent(List<SmellType> issues)
             //TODO: Contact ContentService 
             return result;
         }
+
+        public NodeProgress BuildNodeProgressForTrainee(Trainee trainee, KnowledgeNode knowledgeNode)
+        {
+            var learningObjects = new List<LearningObject>();
+            var sortedPreferences = trainee.LearningPreferenceScore().OrderByDescending(key => key.Value).ToList();
+
+            foreach (var summary in knowledgeNode.LearningObjectSummaries)
+            {
+                foreach (var learningObject in sortedPreferences
+                    .Select(preference => GetLearningObjectForPreference(preference.Key, summary.Id))
+                    .Where(learningObject => learningObject != null))
+                {
+                    learningObjects.Add(learningObject);
+                    break;
+                }
+            }
+
+            return new NodeProgress
+            {
+                Trainee = trainee, Node = knowledgeNode, Status = NodeStatus.Started, LearningObjects = learningObjects
+            };
+        }
+
+        private LearningObject GetLearningObjectForPreference(LearningPreference learningPreference, int summaryId)
+        {
+            return learningPreference switch
+            {
+                LearningPreference.Aural => _learningObjectRepository.GetVideoForSummary(summaryId),
+                LearningPreference.Kinaesthetic => _learningObjectRepository.GetQuestionForSummary(summaryId),
```

Comment:
ArrangeTask is also an interactive LO. I would change the repo function to GetInteractiveLOForSummary.